hydra-node/src/Hydra/Ledger/Simple.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- | A mock implementation of a ledger slightly less dumb than 'Mock'.
+--
+-- This implementation of a 'Ledger' adds a bit more logic in order to:
+-- * Be able to have a representation of 'UTxO' closer to what a real-life eUTxO would be,
+--   so that we can distinguish it from other components of the ledger,
+-- * Be able to have transactions validated against the current state of the ledger, so that
+--   one transaction could at some point be invalid, then becomes valid because some inputs it
+--   consumes is now available.
+module Hydra.Ledger.Simple where
+
+import Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import Cardano.Prelude
+import qualified Data.Set as Set
+import Hydra.Ledger
+
+-- | Simple transaction.
+-- A transaction is a 'TxId', a list of inputs and a list of outputs.
+data SimpleTx = SimpleTx
+  { txId :: TxId
+  , txInputs :: Set TxIn
+  , txOutputs :: Set TxIn
+  }
+  deriving stock (Eq, Ord, Generic, Read, Show)
+
+instance ToCBOR SimpleTx where
+  toCBOR (SimpleTx txid inputs outputs) =
+    toCBOR txid <> toCBOR inputs <> toCBOR outputs
+
+instance FromCBOR SimpleTx where
+  fromCBOR = SimpleTx <$> fromCBOR <*> fromCBOR <*> fromCBOR
+
+type TxId = Integer
+
+-- |An identifier for a single output of a 'SimpleTx'.
+newtype TxIn = TxIn Integer
```

Comment:
Given the docstring, wouldn't `TxOut` fit better?

Reply:
That's debatable yes. I think I followed Cardano's "tradition" of using the same type for inputs and outputs, and leaning towards the former. As said above, I will remove these files from the PR and we can discuss them later on.